src/table/thead.tsx

```diff
@@ -133,27 +131,10 @@ const Thead = React.forwardRef(
 
           {columnDefinitions.map((column, colIndex) => {
             const columnId = getColumnKey(column, colIndex);
-
-            let widthOverride;
-            if (resizableColumns) {
-              if (columnWidths) {
-                // use stateful value if available
-                widthOverride = columnWidths[columnId];
-              }
-              if (colIndex === columnDefinitions.length - 1 && containerWidth && containerWidth > totalWidth) {
-                // let the last column grow and fill the container width
-                widthOverride = 'auto';
-              }
-            }
             return (
               <TableHeaderCell
                 key={columnId}
                 className={headerCellClass}
-                style={{
```

Comment:
This part is taken care of by the column widths util. The exact moment when the width is set in the element lifecycle is now slightly different so having all widths set the same way reduces the risk of the properties being out of sync at any given point in time.

Reply:
Update: this part is still needed as otherwise the initial rendering performance is impacted: every update to the styles of rendered nodes issues a reflow.